### ISSUE TITLE
Process for Kubernetes upgrades?

### ISSUE THREAD

**Issue by vrmerlin** [puppetlabs/puppetlabs-kubernetes#364]
## Describe the Change You Would Like
A clear and concise description of what you would like changed.

We are currently using the Kubernetes Puppet module version 3.2.2, and we'd like to upgrade to 4.0.0. Is there a documented procedure for doing this upgrade cleanly? Do we need to re-create the entire infrastructure with a new YAML file created from kubetool? That sounds like a big impact, and I'm hoping there is another way.

**Comment by vrmerlin** [comment]
I posted this well over a year ago, and never received a reply. Would someone please respond? It seems to me this would be an important issue for many people.

**Comment by puetzp** [comment]
I started to use this module at version 3.x.x as well, now running 6.3.0. I think you would only need to run `kubetool` again if any of the certificates generated by `kubetool` expired in the meantime. 
The only other parameters that I changed frequently during the lifetime of a cluster were `kubernetes::token`, `kubernetes::kubernetes_version` and `kubernetes::kubernetes_package_version`. But this can be done in Hiera directly, no need to re-run `kubetool`.

So my understanding is: Run `kubetool` once for the initial cluster setup or when `kubetool`-generated certificates expire. Modify other Hiera params by hand when the situation demands it.
